### ISSUE TITLE
Google SSO not working

### ISSUE THREAD

**Issue by MMore** [kviklet/kviklet#106]
Hey,

just did the setup for Google SSO. After the redirect from Google, I landed on the Open Requests screen with a loading spinner. Everywhere where I click, I get a loading spinner and/or no data. The console reveals that all api requests fail with `{"message":"Access denied"}`. But I am kinda logged in as I can at least click through the settings options for instance.

Server logs shows
```
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]2024-05-13T12:53:21.569Z ERROR 319 --- [io-8080-exec-10] d.k.k.c.ExceptionHandlerController       : Access denied at /connections/
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]org.springframework.security.access.AccessDeniedException: Access Denied
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at dev.kviklet.kviklet.security.AuthorizationManagerInterceptor.attemptPreAuthorization(MethodSecurityConfiguration.kt:205) ~[classes!/:0.0.1-SNAPSHOT]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at dev.kviklet.kviklet.security.AuthorizationManagerInterceptor.invoke(MethodSecurityConfiguration.kt:161) ~[classes!/:0.0.1-SNAPSHOT]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:756) ~[spring-aop-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123) ~[spring-tx-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:391) ~[spring-tx-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119) ~[spring-tx-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:756) ~[spring-aop-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708) ~[spring-aop-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at dev.kviklet.kviklet.service.ConnectionService$$SpringCGLIB$$0.listConnections(<generated>) ~[classes!/:0.0.1-SNAPSHOT]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at dev.kviklet.kviklet.controller.ConnectionController.getDatasourceConnections(ConnectionController.kt:237) ~[classes!/:0.0.1-SNAPSHOT]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[na:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[na:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:343) ~[spring-aop-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196) ~[spring-aop-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:756) ~[spring-aop-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.validation.beanvalidation.MethodValidationInterceptor.invoke(MethodValidationInterceptor.java:141) ~[spring-context-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:756) ~[spring-aop-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708) ~[spring-aop-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at dev.kviklet.kviklet.controller.ConnectionController$$SpringCGLIB$$0.getDatasourceConnections(<generated>) ~[classes!/:0.0.1-SNAPSHOT]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[na:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[na:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:118) ~[spring-webmvc-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:884) ~[spring-webmvc-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:797) ~[spring-webmvc-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1081) ~[spring-webmvc-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:974) ~[spring-webmvc-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1011) ~[spring-webmvc-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:903) ~[spring-webmvc-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:564) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885) ~[spring-webmvc-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:205) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) ~[tomcat-embed-websocket-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy.lambda$doFilterInternal$3(FilterChainProxy.java:231) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:365) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.access.intercept.AuthorizationFilter.doFilter(AuthorizationFilter.java:100) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:126) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:120) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:131) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:85) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:100) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:179) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:227) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:221) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.oauth2.client.web.OAuth2AuthorizationRequestRedirectFilter.doFilterInternal(OAuth2AuthorizationRequestRedirectFilter.java:181) ~[spring-security-oauth2-client-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:107) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:93) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:91) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:90) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:75) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.context.SecurityContextHolderFilter.doFilter(SecurityContextHolderFilter.java:82) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.context.SecurityContextHolderFilter.doFilter(SecurityContextHolderFilter.java:69) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:62) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.ForwardedHeaderFilter.doFilterInternal(ForwardedHeaderFilter.java:153) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.session.DisableEncodeUrlFilter.doFilterInternal(DisableEncodeUrlFilter.java:42) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:374) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:233) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:191) ~[spring-security-web-6.1.2.jar!/:6.1.2]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:352) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:268) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.session.web.http.SessionRepositoryFilter.doFilterInternal(SessionRepositoryFilter.java:143) ~[spring-session-core-3.1.1.jar!/:3.1.1]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.session.web.http.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:82) ~[spring-session-core-3.1.1.jar!/:3.1.1]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:352) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:268) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.11.jar!/:6.0.11]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:166) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:482) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:676) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:341) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:391) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:894) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1740) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-embed-core-10.1.11.jar!/:na]
2024-05-13T12:53:21Z app[683ddd1bde3198] fra [info]	at java.base/java.lang.Thread.run(Thread.java:1583) ~[na:na]
```

PS: To make Google work with flycast addresses I set a CNAME in DNS for this address. I can open the page with the new domain as usual in browser. Login not via Google works.

**Comment by Askir** [comment]
If you login via Google the first time this creates a new user. This user likely doesn't have any role yet which is why it can't see anything and everything keeps "loading".
You can give this user a role under settings->users with the existing admin account and then login again and it should load.

Sorry this experience is still really rough with no good error messages, I'm working on better UX for this. If the permissions are not the issue let me know I'll have a more detailed look.

**Comment by Askir** [comment]
So if you don't want to use the username/password login you can just give your Google User the admin role and do future permissions through that.

I'm also adding the option for a default role at some point but maybe you don't want everyone in your organization to have access immediately so I think I went with no roles and therefore permissions as default.

**Comment by MMore** [comment]
Thanks for your reply. I am signing in with the user that is already created. So it's basically the same email like the initial username. So I was expecting that it just used the existing user.

**Comment by Askir** [comment]
SSO and password users are handled separately, so you can have two user with the same email one for sso and one for username/password.

SSO users also can't add a password so they always have to login via SSO. Since you probably only use one or the other (hopefully SSO and just password for initial setup) I didn't really consider this a problem .

My recommendation is to just use SSO, less manual user management and no chance to leak a password.

What would have been the behaviour you expected though? A link to the existing user? And then both password and SSO login working for that account?

I guess that's a somewhat more intuitive behaviour. I was mostly focused on giving the possibility of enforcing SSO logins so that users can't set up a password and circumvent the central user management that way so far.

**Comment by Askir** [comment]
I think I've fixed most of the complaints from here in 0.4 😄 :
- There should no longer be users with the same email
- There is a default role with read permissions so that new users can see requests (not create) and don't stare at a blank page (you can edit these permissions to your liking).
- And there are a few more error notifications if things go wrong.
